src/source/Signaling/LwsApiCalls.c

```diff
@@ -898,7 +898,8 @@ STATUS getChannelEndpointLws(PSignalingClient pSignalingClient, UINT64 time)
     resultLen = pLwsCallInfo->callInfo.responseDataLen;
 
     // Early return if we have a non-success result
-    CHK((SERVICE_CALL_RESULT) ATOMIC_LOAD(&pSignalingClient->result) == SERVICE_CALL_RESULT_OK && resultLen != 0 && pResponseStr != NULL, retStatus);
+    CHK((SERVICE_CALL_RESULT) ATOMIC_LOAD(&pSignalingClient->result) == SERVICE_CALL_RESULT_OK && resultLen != 0 && pResponseStr != NULL,
```

Comment:
Why do you need to return the error here? All these backend API calls return success even if the call to the backend fails itself to ensure the higher-level state machine iterates again and retries. The higher-level state machine uses the response to check for the retry or move to the next state

Reply:
I received a 403 back from the call to `/getSignalingChannelEndpoint` (due to lacking access permissions in IAM in my setup). Since a success code assumes the `pSignalingClient` will have `channelEndpointWss` and `channelEndpointHttps` are set, this will lead to failure later when saving to cache here: https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/master/src/source/Signaling/FileCache.c#L180.

This is misleading the user to think that issue is with the cache rather than a permission error from the API call. The state machine won't progress anyway on a non-zero status due to this line: https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/master/src/source/Signaling/StateMachine.c#L490

Reply:
This is similar to the issue #660 

The main issue is that the caching assumes the result of STATUS_SUCCESS as an indicator of an actual API call success. The caching layer needs to be updated to make a proper distinction and only cache when the backend API call succeeds.

Reply:
Not sure about what you mean by caching layer? It currently reads the `retStatus` to make decision to save to cache or not by calling `signalingCacheSaveToFile`, the only way I could see to make that distinction is by creating another status code (something like `STATUS_API_ERROR`).

Reply:
The fix should be similar to the state machine processing. If the status is OK and we have the data we need then we should proceed with caching. I think this should call for a proper fix. Not sure what the priority is. Perhaps AWS folks could comment.

Reply:
@Sean-Der looks like you are the original author of this, suggestions on this?

Personally I'd prefer this to be explicit, even if it means returning a `STATUS_SUCCESS` here instead of `retStatus`.

Reply:
These were conscious choices to differentiate between critical failures caused from the client side driven by retStatus and anything that could actually be related to the backend. An alternative could have been to flow everything via mapping to retStatus and have a retry logic working only based on the retStatus.